### PR TITLE
Add test to check python Warnings

### PR DESCRIPTION
**Pull request by gpotter2**
This PR is an attempt of adding a check of scapy’s garbage disposal ==> in order to detect memory leaks.

- currently, we trigger all Warnings and fail on it. This includes ResourceWarning, DeprecationWarning, SyntaxWarning

fixes https://github.com/secdev/scapy/issues/1516

### CODE REVIEW

**Comment by gpotter2** [comment]
I’m slowly fixing tons of issues and warnings

**Comment by gpotter2** [comment]
@guedou Should be ready for review. I couldn't find a way of removing the last ResourceWarning

File: scapy/asn1/asn1.py [hunk 1]
```diff
@@ -108,7 +108,7 @@ def __init__(self, key, value, context=None, codec=None):
             codec = {}
         self._codec = codec
 
-    def clone(self):  # /!\ not a real deep copy. self.codec is shared
```
Comment (guedou) [inline scapy/asn1/asn1.py]: Are these changes needed?

File: scapy/config.py [hunk 2]
```diff
@@ -329,6 +330,22 @@ def __set__(self, obj, val):
         obj._logLevel = val
 
 
+def _version_checker(module, minver):
```
Comment (guedou) [inline scapy/config.py]: Why do you need this in this PR?

File: scapy/tools/UTscapy.py [hunk 3]
```diff
@@ -973,4 +969,17 @@ def main(argv):
 
 
 if __name__ == "__main__":
-    sys.exit(main(sys.argv[1:]))
+    if sys.warnoptions:
+        with warnings.catch_warnings(record=True) as cw:
+            warnings.resetwarnings()
+            # Let's discover the garbage waste
+            warnings.simplefilter('error')
+            # TODO fix this: scapy has too many StopIteration (deprecated)
```
Comment (guedou) [inline scapy/tools/UTscapy.py]: s/scapy/Scapy/g

File: scapy/asn1/asn1.py (hunk 1 above)
Comment (gpotter2) [reply scapy/asn1/asn1.py]: This PR does not do only ResourceWarning anymore, but all sort of warnings.

`\ ` is detected as an unknown character escape, and thus raises a SyntaxWarning (becomes a syntaxError with the -W option)

File: scapy/config.py (hunk 2 above)
Comment (gpotter2) [reply scapy/config.py]: Importing distutils is creating a DeprecationWarning (imports imp). That avoids importing it, by copying the LooseVersion behavior

**Comment by gpotter2** [comment]
@guedou @p-l- Also ready for review but this one might take some more time

File: scapy/config.py (hunk 2 above)
Comment (guedou) [reply scapy/config.py]: Could you add some specific unit tests for this function?
